test/document_stores/test_weaviate.py

```diff
@@ -144,3 +144,46 @@ def test_deleting_by_id_or_by_filters(document_store_with_docs):
     document_store_with_docs.delete_documents(filters={"name": ["filename2"]})
     document_store_with_docs.get_all_documents.assert_called()
     assert document_store_with_docs.get_document_count() == 1
+
+
+def get_document_store_with_docs_and_similarity(tmp_path, similarity, recreate_index):
+    document_store = get_document_store(
+        "weaviate", tmp_path=tmp_path, similarity=similarity, recreate_index=recreate_index
+    )
+    document_store.write_documents(DOCUMENTS_XS)
+    yield document_store
+    document_store.delete_index(document_store.index)
+
+
+@pytest.mark.weaviate
+@pytest.mark.parametrize("similarity", ["cosine", "l2", "dot_product"])
+def test_similarity(tmp_path, similarity):
```

Comment:
Hi @bogdankostic , unfortunately the `test_similiarity_score_dot_product` fails for Weaviate, as the scores don't seem to align with what is expected. 
<img width="508" alt="image" src="https://user-images.githubusercontent.com/3887295/183917390-13b47b47-a470-4ea1-9017-8b2a45bea7e2.png">

Any ideas?

Reply:
The reason for this seems to be that Weaviate returns the **negative** dot product, while the other document stores return the "standard" dot product. To have uniform scores between all document stores, we should take the negative of the value returned by Weaviate.

Reply:
Also, I noticed that we need to adapt this [line](https://github.com/zoltan-fedor/haystack/blob/82f1ec5c113f118b78d28fe7554d1d93f25a0ae7/haystack/document_stores/weaviate.py#L704) to use `distance` instead of `certainty` when not using `cosine` as similarity metric.

Reply:
Hah, yes, I have already found that and fixed that. It is already in the commit I am preparing. Thanks!